### DISCUSSION TITLE
How to get the glyph outline contours?

### DISCUSSION THREAD

**Discussion by rodolforg** [harfbuzz/harfbuzz#3062] [Q&A] [answered]
Hi.
I intend to use harfbuzz to finally provide proper support to arabic (and other) languages for Text elements in a vector animation software (Synfig Studio).
I know how to retrieve and draw the text glyphs with FreeType library: it gives me the contour points (vertices and control points) for each glyph. I read the basics of Harfbuzz and I searched the API, but I couldn't found how could I get both info with Harfbuzz.

I only saw https://harfbuzz.github.io/harfbuzz-hb-font.html#hb-font-get-glyph-contour-point , but I didn't try it yet. However, I didn't find out how to get the possible control points for bézier (conic/cubic) curves.

Is there any way harfbuzz provide it? Or should I mix harfbuzz <-> freetype API to succeed?

**Comment by behdad** [comment, marked as answer]
Generally we don't support that; you should combine with FreeType. Or better yet, use libraqm.

That said, there's experimental API in HarfBuzz to fetch glyph outlines. That's disabled by default and the API will change before becoming stable.

cc @ebraminio

**Comment by ebraminio** [comment]
@rodolforg the API is now published publicly and guess you can go simplify your integration if needed https://github.com/harfbuzz/harfbuzz/blob/main/src/hb-draw.h but there are cases like font rendering hinting that freetype can handle better if is needed.
